sdk/src/main/java/io/radar/sdk/util/RadarLogBuffer.kt

```diff
@@ -0,0 +1,245 @@
+package io.radar.sdk.util
+
+import android.content.Context
+import androidx.annotation.VisibleForTesting
+import io.radar.sdk.Radar
+import io.radar.sdk.model.RadarLog
+import org.json.JSONObject
+import java.io.File
+import java.io.FileInputStream
+import java.io.FileOutputStream
+import java.util.*
+import java.util.concurrent.ExecutorService
+import java.util.concurrent.Executors
+import java.util.concurrent.locks.ReentrantLock
+
+/**
+ * Wraps the filesystem to support log caching
+ */
+internal class RadarLogBuffer(
+    /**
+     * Application Context
+     */
+    context: Context,
+    /**
+     * Executor for writing files. This is used to ensure that log entries are written to the file in the order that
+     * they are received.
+     */
+    private val executor: ExecutorService = Executors.newSingleThreadExecutor()
+) {
+
+    private companion object {
+
+        /**
+         * Name of the directory in which the logs are stored
+         */
+        const val DIRECTORY = "RadarLogs"
+
+        /**
+         * Maximum number of logs in the buffer.
+         */
+        const val LOG_BUFFER_SIZE_LIMIT = 500
+    }
+
+    /**
+     * The directory where the log files are stored
+     */
+    private val directory = context.getDir(DIRECTORY, Context.MODE_PRIVATE)
+
+    /**
+     * Lock for handling changes to [currentFile]
+     */
+    private val fileLock = ReentrantLock()
+
+    /**
+     * Lock for handling changes to the files within [DIRECTORY]
+     */
+    private val directoryLock = ReentrantLock()
+
+    /**
+     * Contains the name of the current log file. Modifications to this value should occur within a [fileOp]
+     * operation.
+     */
+    private var currentFile: String
+
+    /**
+     * Number of log entries in the current log file. Modifications to this value should occur within a [fileOp]
+     * operation.
+     */
+    var size: Int
+        private set
+
+    init {
+        //First find current file
```

Comment:
minor nit on comments - should we add a preceding space to all of them to follow the kotlin coding convention?

<img width="445" alt="image" src="https://user-images.githubusercontent.com/739611/140990337-3c671159-6946-4d20-8f60-8f144a6ca3b6.png">

https://kotlinlang.org/docs/coding-conventions.html#horizontal-whitespace

Reply:
Ideally we'd just automate this so we don't need to spend brain CPU cycles on this

Reply:
Updated. There are some tools for this. I tested adding ktlint formatter on this (that's what ran to cause some updates to these files - including this fix), but it has some bugs and probably needs some more work to integrate.